src/main/java/formflow/library/FileController.java

```diff
@@ -181,4 +213,117 @@ RedirectView delete(
       return new RedirectView("/error");
     }
   }
+
+  /**
+   * @param httpSession  - The current httpsession
+   * @param submissionId - The submissionId of the file to be downloaded
+   * @param fileId       - The UUID of the file to be downloaded.
+   * @return ON SUCCESS: A ResponseEntity with a status of ok, a header, the contentType, the content size, and a responseBody
+   * that includes the file.
+   * <p>ON FAILURE: A ResponseEntity with a status of forbidden-403 or not found- 404</p>
+   */
+  @GetMapping("/file-download/{submissionId}/{fileId}")
+  public ResponseEntity<StreamingResponseBody> downloadSingleFile(
+      HttpSession httpSession,
+      @PathVariable String submissionId,
+      @PathVariable String fileId
+  ) {
+
+    Optional<UserFile> maybeFile = uploadedFileRepositoryService.findById(UUID.fromString(fileId));
+    if (maybeFile.isEmpty()) {
+      log.error(String.format("File with id %s may have already been deleted", fileId));
+      return ResponseEntity.notFound().build();
+    }
+
+    UserFile file = maybeFile.get();
+    if (!submissionId.equals(httpSession.getAttribute("id").toString())) {
```

Comment:
This would be better as the first thing you do (on line 232) - as you have the data handy and can short circuit any other work if this check fails.  Rather than do some look ups and then check data you didn't have to look up. 

Also would be good to log an error if we fail this check.